src/commands/promote.ts

```diff
@@ -105,102 +102,109 @@ export default class Promote extends Command {
-      for (const arch  of archs) {
+      const arches = buildConfig.targets.filter(t => t.platform === 'win32').map(t => t.arch)
+      await Promise.all(arches.map(async arch => {
         const winPkg = templateShortKey('win32', {bin: config.bin, version: flags.version, sha: flags.sha, arch})
         const winCopySource = cloudBucketCommitKey(winPkg)
         // strip version & sha so scripts can point to a static channel exe
         const unversionedExe = winPkg.replace(`-v${flags.version}-${flags.sha}`, '')
-        const winKey = cloudChannelKey(unversionedExe)
-        // eslint-disable-next-line no-await-in-loop
-        await aws.s3.copyObject(
+        await Promise.all([aws.s3.copyObject(
           {
-            Bucket: s3Config.bucket,
-            ACL: s3Config.acl || 'public-read',
+            ...awsDefaults,
             CopySource: winCopySource,
-            Key: winKey,
-            CacheControl: maxAge,
+            Key: cloudChannelKey(unversionedExe),
           },
-        )
-        // eslint-disable-next-line no-await-in-loop
-        if (flags.indexes) await appendToIndex({...indexDefaults, originalUrl: winCopySource, filename: unversionedExe})
+        )].concat(flags.indexes ? [appendToIndex({...indexDefaults, originalUrl: winCopySource, filename: unversionedExe})] : []))
         CliUx.ux.action.stop('successfully')
-      }
+      }))
     }
 
-    // copy debian artifacts
-    const debArtifacts = [
-      templateShortKey('deb', {bin: config.bin, versionShaRevision: debVersion(buildConfig), arch: 'amd64' as any}),
-      templateShortKey('deb', {bin: config.bin, versionShaRevision: debVersion(buildConfig), arch: 'i386' as any}),
-      'Packages.gz',
-      'Packages.xz',
-      'Packages.bz2',
-      'Release',
-      'InRelease',
-      'Release.gpg',
-    ]
-    if (flags.deb) {
-      this.log(`Promoting debian artifacts to ${flags.channel}`)
-      for (const artifact of debArtifacts) {
-        const debCopySource = cloudBucketCommitKey(`apt/${artifact}`)
-        const debKey = cloudChannelKey(`apt/${artifact}`)
-        // eslint-disable-next-line no-await-in-loop
-        await aws.s3.copyObject(
-          {
-            Bucket: s3Config.bucket,
-            ACL: s3Config.acl || 'public-read',
-            CopySource: debCopySource,
-            Key: debKey,
-            CacheControl: maxAge,
-            MetadataDirective: 'REPLACE',
-          },
+    const promoteDebianAptPackages = async () => {
+      // copy debian artifacts
+      const debArtifacts = [
+        templateShortKey('deb', {bin: config.bin, versionShaRevision: debVersion(buildConfig), arch: 'amd64' as any}),
+        templateShortKey('deb', {bin: config.bin, versionShaRevision: debVersion(buildConfig), arch: 'i386' as any}),
+        'Packages.gz',
+        'Packages.xz',
+        'Packages.bz2',
+        'Release',
+        'InRelease',
+        'Release.gpg',
+      ]
```

Comment:
Is there any reason to have this outside the `if (flags.dev)` statement?
Would be nice to put it inside.

Reply:
all the conditional-on-flag logic moved to the Promise.all like your previous suggestion.
https://github.com/oclif/oclif/pull/1035/files#diff-ecee38a5be3a5fc38b4b9db99fbb0b069663f806030445987ec22835d59a034dR205